### DISCUSSION TITLE
Error when running "cargo leptos watch"

### DISCUSSION THREAD

**Discussion by GreyCipher-sec** [wasm-bindgen/wasm-bindgen#4021] [Q&A] [answered]
### Describe the Bug

I've just created a leptos project with "cargo leptos new --git https://github.com/leptos-rs/start".
The project has been created without problems but when I ran "cargo leptos watch", after the compilation I got this error:

```
PS C:\Users\%USERNAME%\Documents\Coding\leptos-project> cargo leptos watch
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.19s
       Cargo finished cargo build --package=rusty-llama --lib --target-dir=C:\Users\%USERNAME%\Documents\Coding\leptos-project\target\front --target=wasm32-unknown-unknown --no-default-features --features=hydrate
       Front compiling WASM
Error: at `D:\a\cargo-leptos\cargo-leptos\src\compile\front.rs:47:30`

Caused by:
    0: at `D:\a\cargo-leptos\cargo-leptos\src\compile\front.rs:120:10`
    1:

       it looks like the Rust project used to create this wasm file was linked against
       version of wasm-bindgen that uses a different bindgen format than this binary:

         rust wasm file schema version: 0.2.92
            this binary schema version: 0.2.89

       Currently the bindgen format is unstable enough that these two schema versions
       must exactly match. You can accomplish this by either updating this binary or
       the wasm-bindgen dependency in the Rust project.

       You should be able to update the wasm-bindgen dependency with:

           cargo update -p wasm-bindgen --precise 0.2.89

       don't forget to recompile your wasm file! Alternatively, you can update the
       binary with:

           cargo install -f wasm-bindgen-cli --version 0.2.92

       if this warning fails to go away though and you're not sure what to do feel free
       to open an issue at https://github.com/rustwasm/wasm-bindgen/issues!
```

I've executed both the commands to update and install the new version, but when I run again "cargo leptos watch" I get the same output.

### Steps to Reproduce

1. "cargo leptos new --git https://github.com/leptos-rs/start"
2. Enter the project directory
3. "cargo leptos watch"

### Expected Behavior

I should be able to see a website hosted on localhost.

### Actual Behavior

Doesn't compile and get always the same error.

**Comment by kanielrkirby** [comment]
I'm no expert, but it would definitely help anyone wanting to help you to have more information about your environment. What did you use to set up your Rust environment (`rustup` or something else)? Which `wasm` targets have you added (`rustup target list --installed`)? What's your OS / package manager / `cargo` version? 

You might also try removing the `target`, `~/.cargo`, some of the `~/.cache` folders that are related, etc., as sometimes caches can conflict. Just things that have helped me with similar issues.

**Comment by Liamolucko** [comment, marked as answer]
This is an unfortunate consequence of the way Leptos handles wasm-bindgen versions.

The first command doesn't work because Leptos' starter template sets `wasm-bindgen = "=0.2.92"`, and `cargo update` can't update pinned versions.

The second command doesn't work because `cargo leptos` doesn't actually use `wasm-bindgen-cli`: it bakes in a specific version of `wasm-bindgen-cli-support`.

Installing the latest version of `cargo-leptos` should fix this, since it uses `wasm-bindgen` 0.2.92 the same as the template.

See also #3900.
